statistics/2025-04-28-p2ra/index.qmd

```diff
@@ -0,0 +1,1111 @@
+    .by = c(species, group),
+  ) %>%
+  arrange(species)
+
+kable(species_ww_summary)
+```
+
+This code calculates summary statistics for the observed wastewater data, which we'll use to compare with predictions from our model.
+
+```{r}
+posterior_predictive_ww_summary %>%
+  ggplot(aes(x = pos_samples, fill = group, after_stat(density))) +
+  geom_histogram(binwidth = 1, center = 0, color = "grey") +
+  geom_vline(data = species_ww_summary, mapping = aes(xintercept = pos_samples)) +
+  facet_wrap(vars(species)) +
+  scale_x_continuous(breaks = seq(0, 12, 2), name = "positive samples")
+```
+
+This plot compares the observed number of positive wastewater samples (vertical line) with the distribution generated from our model.
+It looks like the model systematically overestimates the number of wastewater samples with viral reads (though not by a lot in any one case).
+This may be a sign that $\phi_{ww}$ is too large: too little overdispersion leading to too few zero counts given a mean relative abundance.
+Or it could be that $\mu_{ww}$ is too slightly high, perhaps because of the prior.
+
+```{r}
+posterior_predictive_ww_summary %>%
+  ggplot(aes(x = viral_reads_total, y = species, fill = group)) +
+  geom_violin(draw_quantiles = c(0.25, 0.5, 0.75)) +
+  geom_point(data = species_ww_summary) +
+  scale_x_log10(
+    breaks = 10^seq(0, 4),
+    limits = c(1, 2e4),
+    name = "total viral read count (conditional on > 0)",
+    oob = scales::oob_keep,
+  ) +
+  scale_y_discrete(limits = rev(species_order))
+```
+
+This plot compares the observed total viral reads in wastewater (points) with the distribution predicted by our model.
+The observations here are all very close to the posterior median.
+I'm not sure if this is a problem or not.
+I'm inclined to think that it's not, given that the posteriors are widely dispersed.
+
+```{r}
+posterior_predictive_ww_summary %>%
+  ggplot(aes(x = viral_reads_max, y = species, fill = group)) +
+  geom_violin(draw_quantiles = c(0.25, 0.5, 0.75)) +
+  geom_point(data = species_ww_summary) +
+  scale_x_log10(
+    breaks = 10^seq(0, 4),
+    limits = c(1, 2e4),
+    name = "largest viral read count (conditional on > 0)",
+    oob = scales::oob_keep,
+  ) +
+  scale_y_discrete(limits = rev(species_order))
+```
+
+This plot compares the observed maximum viral reads in wastewater with the distribution from our model.
+
+Here the model might be systematically slightly overestimating the maximum viral reads.
+This cuts against the "too little overdispersion" hypothesis for the excess positive samples.
+Perhaps $\mu_{ww}$ is slightly too high, perhaps because the prior is pushing the estimates up?
+
+```{r}
+posterior_predictive_ww_summary %>%
+  ggplot(aes(x = viral_reads_sd, y = species, fill = group)) +
+  geom_violin(draw_quantiles = c(0.25, 0.5, 0.75)) +
+  geom_point(data = species_ww_summary) +
+  scale_x_log10(
+    breaks = 10^seq(0, 4),
+    name = "stdev viral read count (conditional on > 0)",
```

Comment:
But when we compare mu_ww prior to posterior (page 27) it doesn't seem like the priors are chronically left or right of the posteriors. Wouldn't we expect posterior to be consistently lower than prior if prior were too high?